server/lib/parties/party-service.ts

```diff
@@ -0,0 +1,217 @@
+import cuid from 'cuid'
+import { Map, Record } from 'immutable'
+import { NydusServer } from 'nydus'
+import { singleton } from 'tsyringe'
+import {
+  ClientSocketsGroup,
+  ClientSocketsManager,
+  UserSocketsGroup,
+  UserSocketsManager,
+} from '../websockets/socket-groups'
+
+const MAX_PARTY_SIZE = 8
+
+export class PartyUser extends Record({
+  id: null as number | null,
+  name: null as string | null,
+}) {}
+
+export class PartyRecord extends Record({
+  id: null as string | null,
+  invites: Map<number | null, PartyUser>(),
+  members: Map<number | null, PartyUser>(),
+  leader: null as number | null,
+}) {}
+
+enum ErrorEnum {
+  Unauthorized = 'Unauthorized',
```

Comment:
If/when you move away from directly mirroring http-error codes, I'd probably avoid a String enum (just let typescript assign number values)